python/grass/tools/session_tools.py

```diff
@@ -390,12 +446,42 @@ def run_cmd(
         :param tool_kwargs: named tool arguments used for error reporting (experimental)
         :param **popen_options: additional options for :py:func:`subprocess.Popen`
         """
-        return self.call_cmd(
-            command,
-            tool_kwargs=tool_kwargs,
-            input=input,
-            **popen_options,
-        )
+        # Compute the environment for subprocesses and store it for later use.
+        if "env" not in popen_options:
+            popen_options["env"] = self._modified_env_if_needed()
+
+        if parameter_converter is None:
+            parameter_converter = ParameterConverter()
+            parameter_converter.process_parameter_list(command[1:])
+        try:
+            if parameter_converter.import_export:
+                if self._importer_exporter is None:
+                    self._importer_exporter = ImporterExporter(
+                        run_function=self.call, run_cmd_function=self.call_cmd
+                    )
+                command = self._importer_exporter.process_parameter_list(
+                    command, **popen_options
+                )
+                self._importer_exporter.import_data(env=popen_options["env"])
+            # We approximate tool_kwargs as original kwargs.
```

Comment:
Maybe explain this more

Reply:
I settled in the code to just considering the original kwargs what we want to pass for the error reporting. This may or may not change, but in any case, now this is clearly internal code.

Let me know if I need to add more comments here or about anything else.